### DISCUSSION TITLE
Getting malloc failed on test.wasm(Hello world)

### DISCUSSION THREAD

**Discussion by sumyrai** [bytecodealliance/wasm-micro-runtime#2675] [Q&A]
Executing Test.wasm on iwasm Ubuntu 22.04 x64 returns next: ' [05:58:26:281 - 7F668EFEA740]: warning: allocate 16 bytes memory failed. Exception: allocate memory failed'

**Comment by wenyongh** [comment]
Hi, please try adding `--heap-size=n` for iwasm, e.g., `iwasm --heap-size=32768 ..`.

**Comment by nerdbugg** [comment]
Still get same error on ubuntu 20.04 && archlinux. Is there some way to find out why it failed?

**Reply by nerdbugg** [reply]
I attempted to use `/opt/wasi-sdk/bin/clang -O3 -o test.wasm main.c` to recompile `test.wasm`. And it run successfully.

**Reply by nerdbugg** [reply]
The reason maybe the option in `build.sh`: `-nostdlib`. And wamr side do not handle it properly.

**Reply by mint-in-wonderland** [reply]
Many thanks for the details, they successfully addressed the issue.
In addition, the issue still exists on July 25, 2025.
